Tests/Tactics/ReduceFetchInst.lean

```diff
@@ -9,14 +9,50 @@ import Tests.SHA2.SHA512ProgramTest
 
 /-! ## Tests for `refuceFetchInst` simproc -/
 
-example (h : s.program = GCMGmultV8Program.gcm_gmult_v8_program) :
-    fetch_inst 0x7d8800#64 s = (some 1279294481#32) := by
+/--
+error: unsolved goals
+s : ArmState
+z : Option (BitVec 32)
+h : s.program = GCMGmultV8Program.gcm_gmult_v8_program
+⊢ some 1279294481#32 = z
+-/
+#guard_msgs in example (h : s.program = GCMGmultV8Program.gcm_gmult_v8_program) :
+    fetch_inst 0x7d8800#64 s = z := by
   simp (config := {ground := false}) only [reduceFetchInst]
```

Comment:
```suggestion
    fetch_inst 0x7d8800#64 s = z := by
  simp (config := {ground := false}) only [reduceFetchInst]
  trace_state
  all_goals sorry
```

Lean uses [the pattern](https://github.com/leanprover/lean4/blob/6bc98af67b081b2b22b8fc887626f182585bc4bf/tests/lean/run/grind_pre.lean#L25-L27).

I'd prefer we use the same pattern as upstream lean, since I feel it communicates intent a tad better.

Repeat the same for all examples below.

Reply:
Cool, I hadn't seen that yet, and agree that it looks nicer!